### PR TITLE
[ONNX Editor] Use port indices instead of tensor names, new naming convention of inputs

### PR DESCRIPTION
**Pull request by mbencer**
Ticket: 52743, 54016, 54933

- Introduced a new method of edge indication - by node index and input/output port index instead of node index and tensor name
- Changed way of creating new inputs: in general: cutting one edge causes the creation of new input (in general)
- Introduced a new naming convention for new inputs:
-> the original name of an input tensor, if the tensor is consumed by only a one node
-> first output name of an input tensor consumer + "/placeholder_port_" + port_index, if the tensor is consumed by more than one node (compatible with MO)
- Introduce a mechanism to provide custom input names

### CODE REVIEW

File: ngraph/frontend/onnx_editor/src/edge_mapper.cpp [hunk 1]
```diff
@@ -61,36 +61,47 @@ std::vector<int> onnx_editor::EdgeMapper::find_node_indexes(const std::string& n
...
+                                                 const std::string& output_name) const
 {
     if (node_index >= static_cast<int>(m_node_outputs.size()))
     {
         throw ngraph_error("Node with index: " + std::to_string(node_index) +
                            "is out of scope outputs list");
     }
-    if (output_index >= static_cast<int>(m_node_outputs[node_index].size()))
+    const auto out_port_idx = std::find(
+        std::begin(m_node_outputs[node_index]), std::end(m_node_outputs[node_index]), output_name);
```
Comment (tomdol) [inline ngraph/frontend/onnx_editor/src/edge_mapper.cpp]: Possible memory violation if node_index is out of bounds. I suggest extracting the `m_node_outputs[node_index]` expression above and assigning to a variable with a pre-check if the index value.

File: ngraph/frontend/onnx_editor/src/edge_mapper.cpp [hunk 2]
```diff
@@ -61,36 +61,47 @@ std::vector<int> onnx_editor::EdgeMapper::find_node_indexes(const std::string& n
...
 {
-    if (node_index >= static_cast<int>(m_node_inputs.size()))
+    if (node_index >= 0 && node_index >= static_cast<int>(m_node_inputs.size()))
     {
         throw ngraph_error("Node with index: " + std::to_string(node_index) +
                            "is out of scope inputs list");
     }
-    if (input_index >= static_cast<int>(m_node_inputs[node_index].size()))
+    const auto matched_inputs = std::count(
+        std::begin(m_node_inputs[node_index]), std::end(m_node_inputs[node_index]), input_name);
```
Comment (tomdol) [inline ngraph/frontend/onnx_editor/src/edge_mapper.cpp]: Similarily to the comment above - possible access violation.

File: ngraph/frontend/onnx_editor/src/edge_mapper.cpp [hunk 3]
```diff
@@ -61,36 +61,47 @@ std::vector<int> onnx_editor::EdgeMapper::find_node_indexes(const std::string& n
...
-    const auto input_name = m_node_inputs[node_index][input_index];
-    return input_name;
+    if (matched_inputs > 1) // more indexes with the same name
+    {
+        throw ngraph_error("Node with index: " + std::to_string(node_index) +
+                           " has more than one inputs with name: " + input_name +
+                           ". You should use port indexes to distinguish them.");
+    }
+    const auto in_port_idx = std::find(
+        std::begin(m_node_inputs[node_index]), std::end(m_node_inputs[node_index]), input_name);
```
Comment (tomdol) [inline ngraph/frontend/onnx_editor/src/edge_mapper.cpp]: Here as well, you can reuse a variable if you assign `m_node_inputs[node_index]` to it somewhere above.

File: ngraph/frontend/onnx_editor/src/detail/subgraph_extraction.cpp [hunk 4]
```diff
@@ -99,20 +99,49 @@ namespace
...
+    std::string get_input_tensor_name(ONNX_NAMESPACE::GraphProto& graph, const InputEdge& edge)
+    {
+        validate_node_index(graph, edge.m_node_idx);
+        NGRAPH_CHECK(edge.m_port_idx >= 0 &&
+                         edge.m_port_idx <
+                             static_cast<int>(graph.node(edge.m_node_idx).input().size()),
+                     "Node with index: ",
+                     std::to_string(edge.m_node_idx),
+                     " has not input with index: ",
+                     std::to_string(edge.m_port_idx));
```
Comment (tsocha) [inline ngraph/frontend/onnx_editor/src/detail/subgraph_extraction.cpp]: Could you add a helper for validating ports?
It seems odd that node_index have a helper for it but port doesn't.

File: ngraph/frontend/onnx_editor/src/detail/subgraph_extraction.cpp [hunk 5]
```diff
@@ -99,20 +99,49 @@ namespace
...
+    std::string get_output_tensor_name(ONNX_NAMESPACE::GraphProto& graph, const OutputEdge& edge)
+    {
+        validate_node_index(graph, edge.m_node_idx);
+        NGRAPH_CHECK(edge.m_port_idx >= 0 &&
+                         edge.m_port_idx <
+                             static_cast<int>(graph.node(edge.m_node_idx).output().size()),
+                     "Node with index: ",
+                     std::to_string(edge.m_node_idx),
+                     " has not output with index: ",
+                     std::to_string(edge.m_port_idx));
```
Comment (tsocha) [inline ngraph/frontend/onnx_editor/src/detail/subgraph_extraction.cpp]: /\

File: ngraph/frontend/onnx_editor/src/detail/subgraph_extraction.cpp (hunk 4 above)
Comment (mbencer) [reply ngraph/frontend/onnx_editor/src/detail/subgraph_extraction.cpp]: done

File: ngraph/frontend/onnx_editor/src/detail/subgraph_extraction.cpp (hunk 5 above)
Comment (mbencer) [reply ngraph/frontend/onnx_editor/src/detail/subgraph_extraction.cpp]: as above

File: ngraph/frontend/onnx_editor/src/edge_mapper.cpp (hunk 1 above)
Comment (mbencer) [reply ngraph/frontend/onnx_editor/src/edge_mapper.cpp]: Checking if node_index is out of bounds is in line 67, but extraction to variable seems to be a good idea. On the occasion checking change to `NGRAPH_CHECK` way.

File: ngraph/frontend/onnx_editor/src/edge_mapper.cpp (hunk 2 above)
Comment (mbencer) [reply ngraph/frontend/onnx_editor/src/edge_mapper.cpp]: as above

File: ngraph/frontend/onnx_editor/src/edge_mapper.cpp (hunk 3 above)
Comment (mbencer) [reply ngraph/frontend/onnx_editor/src/edge_mapper.cpp]: done

**Pull request closed without merging**